cpp/src/arrow/filesystem/gcsfs_test.cc

```diff
@@ -286,6 +286,20 @@ TEST_F(GcsIntegrationTest, GetFileInfoObject) {
   arrow::fs::AssertFileInfo(fs.get(), PreexistingObjectPath(), FileType::File);
 }
 
+TEST_F(GcsIntegrationTest, CopyFileSuccess) {
+  auto fs = internal::MakeGcsFileSystemForTest(TestGcsOptions());
+  const auto destination_path = kPreexistingBucket + std::string("/copy-destination");
+  ASSERT_OK(fs->CopyFile(PreexistingObjectPath(), destination_path));
+  arrow::fs::AssertFileInfo(fs.get(), destination_path, FileType::File);
+}
+
+TEST_F(GcsIntegrationTest, CopyFileNotFound) {
+  auto fs = internal::MakeGcsFileSystemForTest(TestGcsOptions());
+  const auto destination_path = kPreexistingBucket + std::string("/copy-destination");
+  const auto status = fs->CopyFile(NotFoundObjectPath(), destination_path);
+  EXPECT_EQ(status.code(), StatusCode::IOError);
```

Comment:
Nit: this can be shortened to:
```c++
ASSERT_RAISES(IOError, fs->CopyFile(NotFoundObjectPath(), destination_path));
```

If the error message is predictable enough, you could also test for it using e.g.:
```c++
EXPECT_RAISES_WITH_MESSAGE_THAT(IOError, ::testing::HasSubstr("xxx"),
    fs->CopyFile(NotFoundObjectPath(), destination_path));
```

Reply:
Done. PTAL.